app/addons/documents/index-results/stores.js

```diff
@@ -176,33 +195,31 @@ function (app, FauxtonAPI, ActionTypes, HeaderActionTypes, Documents, MangoHelpe
         isDeletable: this.isDeletable(doc),
         isEditable: this.isEditable(doc)
       };
-
     },
 
-    filterOutGeneratedMangoDocs: function (doc) {
-      if (doc.get && typeof doc.get === 'function') {
-        return doc.get('language') !== 'query';
-      }
+    // we need to cache the tableview data and manage the state inside the store as we
+    // are able to select and deselect fields using the dropdown which would then lazyly
+    // repopulate the colelctions on a store emit, which is perfectly fine, however we run
+    // into race conditions with other components which also listen to the same store.
```

Comment:
this is super annoying, the problem here is that

1. action requests data, gets data
2. puts into store, store emits change event
3. the change events leads to components pulling the new data
4. the new components render
5 the data was not processed fully yet, (e.g. the schema calculation) 
6 old state is displayed and errors occur, as components depends on data that is calculated after the data arrived in the store

i thought of and decided against: cascading stores updates (store depends on updates from another store) and ended up with caching, however this introduced some complexity for managing that state (you have to make sure to override it properly every time and don't get old data and so on)

not happy with this, bit also not sure regarding a better approach? maybe redux can help here with the transformations/reductions that are separate from the store